### DISCUSSION TITLE
Batch Size for NER

### DISCUSSION THREAD

**Discussion by morankyle** [explosion/spaCy#8301] [Help: Best practices]
First off thank you! After changing architecture from following Spacy 2.0 to 3.0 saw a 20 point increase in F1-Score! 

My question is about the training batch size for NER, 100 start and 1000 stop seems large, why is the default that high and have you seen success with lesser numbers?

**Comment by svlandeg** [comment]
Great to hear you're seeing such a performance boost when switching to spaCy 3! 20 percent point is quite an unexpectedly huge leap though, but perhaps your dataset isn't very large? Because in that case, larger variations are expected.

With respect to the batch training size, these defaults have simply kind of worked for us when [benchmarking](https://spacy.io/usage/facts-figures#benchmarks) our pretrained pipelines on standard evaluation sets. It's always difficult to recommend good default settings in general, as every use-case and dataset is different. But you can try out some different runs, varying the batch size in the config, and see whether that makes any difference in your specific project. Feel free to report the results here, as I'm sure others may be interested as well!

**Reply by morankyle** [reply]
Ya dataset isn't huge, but definitely getting consistently higher accuracy with 3.0 then 2.0 (~ 20 points). 

For my use case dropping the batch size did not have really any effect other then obviously increasing the time to train. 

Will see how playing with the patience, max_steps, and learning rate effects it as well. Thanks again!

**Reply by morankyle** [reply]
Also another quick note, in its documentation MaxoutWindowEncoder.v2, it says the recommended value for depth is 4 but with the quickstart if you select English -> NER -> CPU -> accuracy, the value given for depth is 8

**Reply by svlandeg** [reply]
Ah, right, so the code for that, behind the quickstart is [here](https://github.com/explosion/spaCy/blob/master/spacy/cli/templates/quickstart_training.jinja):

```
[components.tok2vec.model.encode]
@architectures = "spacy.MaxoutWindowEncoder.v2"
width = {{ 96 if optimize == "efficiency" else 256 }}
depth = {{ 4 if optimize == "efficiency" else 8 }}
```
so you see that it depends on the `efficiency` setting. You picked accuracy, so you're getting 8 ;-)

**Reply by morankyle** [reply]
:) 

Last question (sorry :) ) , and this one is a bit silly but I could replace the MaxoutWindowEncoder with the TorchBiLSTMEncoder in the config file correct? Other than having pytorch installed there would be no other steps?

**Reply by svlandeg** [reply]
Yes, that should work!

**Reply by morankyle** [reply]
TorchBiLSTMEncoder hurt performance but the MishWindowEncoder was amazing! ~6 point boost !

**Reply by svlandeg** [reply]
Ah cool, interesting!
